feat: thirdparty rework (#522)
supertokens/docs: v2/thirdparty/custom-ui/multitenant-login.mdx

---
id: multitenant-login
title: "Multitenant login"
hide_title: true
---

import AppInfoForm from "/src/components/appInfoForm"
import BackendSDKTabs from "/src/components/tabs/BackendSDKTabs";
import GoFrameworkSubTabs from "/src/components/tabs/GoFrameworkSubTabs";
import TabItem from '@theme/TabItem';
import MultitenancyIntro from '../reusableMD/multitenancy/intro.mdx'
import CreateConfigureTenants from '../reusableMD/multitenancy/create-configure-tenants.mdx'
import BackendCORS from '../reusableMD/multitenancy/backend-cors.mdx'

:::important
TODO BANNER This is a paid feature

If you wish to use this feature for free, follow this implementation guide TODO link
:::

# Multitenant login

<MultitenancyIntro />

<CreateConfigureTenants />

## Step 2: Frontent setup

To fetch the login methods dynamically from the backend, use the following API:

> ```tsx

Reviewer: Use the right components for this.. see other pages in custom UI section.

> 

Reviewer: ```suggestion
```

import SuperTokens from 'supertokens-web-js';

> 

Reviewer: change this entier page to not use okta, and use AD instead

const loginMethods = await SuperTokens.getLoginMethods({
  tenantId: '...'
})

if (loginMethods.thirdParty.enabled) {
  const providers = loginMethods.thirdParty.providers;

>   // render the login buttons

Reviewer: ```suggestion
  if (providers.find(i => i.thirdPartyId === "okta")) {
     // render sign in with okta button
  }
  // more checks
```

} else {
  // thirdparty login is disabled for the tenant
}
```

> 

Reviewer: mention what happens next -> when you call getAuthorizationUrl function, pass in the tenantId to it. SignInUp function remans the same. This applies for web. For mobile, say that tenantId needs to be passed to sign in up API call as well.

## Step 3: Backend setup

> Remove the `providers` list from the `thirdparty` config. The backend SDK will automatically pick up the configuration stored in the core.
> 
> <BackendSDKTabs>
> <TabItem value="nodejs">
> 
> ```tsx
> TODO
> ```
> </TabItem>
> <TabItem value="go">
> 
> ```go
> import (
> 	"github.com/supertokens/supertokens-golang/recipe/thirdparty"
> )
> 
> func main() {
> 	// Inside supertokens.Init
>   // highlight-start
> 	thirdparty.Init(nil)
> 		// highlight-end
> }
> ```
> </TabItem>
> <TabItem value="python">
> 
> ```python
> TODO
> ```
> 
> </TabItem>
> </BackendSDKTabs>
> 

Reviewer: ```suggestion

```

<BackendCORS />

## See also

> TODO: links

Reviewer: put TODO links instead of this comment

> - Single Login Domain - asking users for tenantId
> - Single Login Domain - redirection post login
> - Subdomain login
> - Getting tenant id of logged in user
> - Allowing end users to add their own SSO provider
> - SAML login

Reviewer: ```suggestion
- [Single domain](../TODO) and [sub domain](../TODO) login
- See all built in multi tenant providers.
- See how to add custom multi tenant provider.
- Getting tenant id of logged in user.
- Allowing end users to add their own provider.
- SAML login.
```
